packages/web-wallet/src/components/Header.tsx

```diff
@@ -0,0 +1,249 @@
+          <div className="flex flex-col md:flex-row md:items-center md:justify-between gap-3">
+            {/* Logo + Badge */}
+            <div className="flex items-center justify-center md:justify-start gap-2">
+              <img
+                src={htrLogo}
+                alt="Hathor"
+                className="h-4 md:h-5"
+              />
+              <div className="px-[6px] py-[6px] bg-transparent border border-white/20 rounded-full flex items-center">
+                <span className="text-[8px] md:text-[9px] font-medium text-white tracking-wider leading-none">WEB WALLET</span>
+              </div>
+            </div>
+
+            {/* Address + Network + Menu */}
+            <div className="flex items-center justify-center gap-2 md:gap-3 md:relative">
+              {/* Wallet Address */}
+              <button
+                onClick={handleCopyAddress}
+                className="px-3 md:px-4 py-2 bg-[#191C21] border border-[#24292F] rounded-full flex items-center gap-1 md:gap-2 hover:bg-[#24292F] transition-colors group"
+              >
+                <span className="text-xs md:text-sm font-mono text-white">
+                  {address ? truncateAddress(address) : 'Not connected'}
+                </span>
+                {address && <Copy className="w-3 h-3 md:w-4 md:h-4 text-muted-foreground group-hover:text-primary transition-colors" />}
+              </button>
+
+              {/* Network Button */}
+              <button
+                onClick={() => setIsNetworkDialogOpen(true)}
+                className="px-3 md:px-5 py-2 bg-[#191C21] border border-[#24292F] rounded-full flex items-center gap-1 md:gap-2 hover:bg-[#24292F] transition-colors"
+              >
+                <Globe className="w-3 h-3 md:w-4 md:h-4 text-white" />
+                <span className="text-xs md:text-sm font-medium text-white">
+                  {getNetworkDisplayName(network)}
+                </span>
+              </button>
+
+              {/* Hamburger Menu Button */}
+              <button
+                onClick={() => setIsMenuOpen(!isMenuOpen)}
+                className="p-2 bg-[#191C21] border border-[#24292F] rounded-full hover:bg-[#24292F] transition-colors"
+              >
+                {isMenuOpen ? (
+                  <X className="w-4 h-4 md:w-5 md:h-5 text-white" />
+                ) : (
+                  <Menu className="w-4 h-4 md:w-5 md:h-5 text-white" />
+                )}
+              </button>
+
+              {/* Desktop: Compact dropdown - only on desktop, stays in relative container */}
+              {isMenuOpen && (
+                <div className="hidden md:block absolute top-full right-0 mt-2 w-48 bg-[#191C21] border border-[#24292F] rounded-lg shadow-lg overflow-hidden z-50">
+                  <button
+                    onClick={() => {
+                      setIsMenuOpen(false);
+                      onCreateTokenClick?.();
```

Comment:
I have never seen optional chaining operator used on functions, can we be sure this works?

Reply:
Yes, it works